omniduct/filesystems/webhdfs.py

```diff
@@ -133,11 +133,18 @@ def _dir(self, path):
                 replication=f['replication']
             )
 
-    def _mkdir(self, path, recursive):
-        raise NotImplementedError
+    def _mkdir(self, path, recursive, exist_ok):
+        if not path.endswith('/'):
+            path += '/'
+        if not self._exists(path):
+            self.__webhdfs.make_dir(path)
 
     def _remove(self, path, recursive):
-        raise NotImplementedError
+        if self._exists(path):
```

Comment:
The existence check is performed by the parent FileSystemClient.remove, and the check for `recursive` is also performed by this method. You should be able to replace this function body with just:
```
self.__webhdfs.delete_file_dir(path, recursive)
```

Reply:
done